src/cuCompactor.cuh

```diff
@@ -46,37 +48,39 @@ __global__ void compactK(T* d_input,int length, T* d_output,int* d_BlocksOffset,
 		int pred = predicate(d_input[idx]);
 		int w_i = threadIdx.x/warpSize; //warp index
 		int w_l = idx % warpSize;//thread index within a warp
-		int t_m = INT_MAX >> (warpSize-w_l-1); //thread mask (ERROR IN THE PAPERminus one is required)
 
-		int b	= __ballot(pred) & t_m; //balres = number whose ith bit isone if the ith's thread pred is true masked up to the current index in warp
+		// compute exclusive prefix sum based on predicate validity to get output offset for thread in warp
+		int t_m = FULL_MASK >> (warpSize-w_l); //thread mask
+		int b	= __ballot_sync(FULL_MASK,pred) & t_m; //balres = number whose ith bit is one if the ith's thread pred is true masked up to the current index in warp
```

Comment:
I need to update this so the code can be compiled with older versions of CUDA. 
Somthing like the following:
```c++
#if (CUDART_VERSION < 9000)
int b	= __ballot(pred) & t_m;
#else
int b	= __ballot_sync(FULL_MASK,pred) & t_m; 
#endif
```

Reply:
Agreed, I will make this change.